models/proposal/proposalReducer.ts

```diff
@@ -1,46 +1,66 @@
 import { MessageMapEntry, ProposalMessageType } from './messageMap'
 import { Proposal } from './proposal'
 import { ProposalAction } from './proposalActions'
-import { topmostId } from './proposalSelectors'
+import { getActiveMessageId, sortedMessages } from './proposalSelectors'
 
+function checkUpdated(updated: Proposal) {
+  if (
+    updated?.activeMessageId &&
+    !updated.messageMap[updated.activeMessageId]
+  ) {
+    console.error(
+      `bad active message in ${JSON.stringify(updated, undefined, 2)}`
+    )
+  }
+  return updated
+}
 export function ProposalReducer(
   state: Proposal,
   action: ProposalAction
 ): Proposal {
   switch (action.type) {
     case 'setTitle':
-      return { ...state, title: action.title }
+      const updated = { ...state, title: action.title }
+      return checkUpdated(updated)
     case 'setDescription':
-      return { ...state, description: action.description }
+      return checkUpdated({ ...state, description: action.description })
     case 'setActiveMessage': {
-      return {
-        ...state,
-        activeMessages: {
-          ...state.activeMessages,
-          [action.messageType]: action.id,
-        },
+      if (state.messageMap[action.id]) {
+        return checkUpdated({
+          ...state,
+          activeMessageId: action.id,
+        })
+      } else {
+        return checkUpdated(state)
       }
     }
     case 'removeMessage': {
+      const currentActiveMessageId = getActiveMessageId(state)
+      let updatedActiveMessageId
       const messageMap = { ...state.messageMap }
-      const existingMessage = state.messageMap[action.id]
-      const messageType = existingMessage.messageType
-      let activeMessageId: string | undefined =
-        state?.activeMessages[messageType]
       delete messageMap[action.id]
-      let activeMessages = state.activeMessages
-      if (action.id === activeMessageId) {
-        activeMessageId = topmostId(messageMap, messageType)
-        activeMessages = {
-          ...activeMessages,
-          [messageType]: activeMessageId,
+      if (action.id === currentActiveMessageId) {
+        const newMessages = sortedMessages(messageMap)
+        if (newMessages?.length) {
+          updatedActiveMessageId = newMessages[0].id
         }
       }
-      return { ...state, messageMap, activeMessages }
+      const updatedProposal: Proposal = {
+        ...state,
+        messageMap,
+        activeMessageId: updatedActiveMessageId ?? currentActiveMessageId,
+      }
+      if (
+        updatedProposal?.activeMessageId &&
+        !updatedProposal.messageMap[updatedProposal.activeMessageId]
+      ) {
+        console.error(`stale active message id`)
+        debugger
```

Comment:
```suggestion
```

Reply:
This is a "should never get here" situation but I removed the debugger statement.